client/engine/engine.go

```diff
@@ -245,8 +247,55 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 			message.SignedStates[0].State(),
 			*e.store.GetAddress(),
 		)
+	case strings.Contains(string(message.ObjectiveId), "Virtual"):
+		initialState := message.SignedStates[0].State()
+
+		if initialState.TurnNum != 0 {
+			return virtualfund.VirtualFundObjective{}, errors.New("cannot construct virtual fund objective without prefund state")
```

Comment:
I think the checks on turn number and number of participants are redundant? They are performed already in the constructor for a virtual channel, and the errors should bubble up and arrive at your L288.

Reply:
Updated in a59c047